q2_mlab/doctor.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python
+import click
+import os
+from datetime import datetime
+
+
+def parse_info(info_filepath):
+    with open(info_filepath) as f:
+        lines = f.readlines()
+        info_dict = {
+            key.strip(): val.strip()
+            for key, val in zip(lines[0].split(","), lines[1].split(","))
+        }
+    return info_dict
+
+
+def get_uninserted_results(results_dir):
+    return [f.name for f in os.scandir(results_dir) if f.path.endswith(".qza")]
+
+
+def get_inserted_results(results_dir):
+    inserted_dir = os.path.join(results_dir, "inserted")
+    if not os.path.isdir(inserted_dir):
+        return []
+    else:
+        return [
+            f.name for f in os.scandir(inserted_dir) if f.path.endswith(".qza")
+        ]
+
+
+def filter_duplicate_parameter_results(
+    list_of_artifact_paths, results_dir, delete=False
```

Comment:
might it be better to ignore duplicates rather than delete them?

Reply:
We ignore duplicates by default, and just omit them from the list of artifact filenames returned. Might never use the option to delete duplicates...but it's there.